### ISSUE TITLE
pipx: Injecting packages fails when using global

### ISSUE THREAD

**Issue by gaby** [ansible-collections/community.general#9009]
### Summary

I took the integration tests from https://github.com/russoz-ansible/community.general/blob/main/tests/integration/targets/pipx/tasks/testcase-injectpkg.yml and try using them with `global: true` but they fail.

This is probably related to the other in which ansible is creating the command to execute with `pipx`. 

The following commands are *not* the same:

```console
pipx list --global
pipx --global list
```

The correct one is the first one, when the playbooks runs it's invoking the 2nd one. Another note, the integration tests from this collection are missing a test for testing for injected global packages.

This is a known issue reported in the following issue:
- https://github.com/pypa/pipx/issues/1443

### Issue Type

Bug Report

### Component Name

pipx

### Ansible Version

```console
ansible [core 2.17.5]
  config file = None
  configured module search path = ['/home/ubuntu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.10/dist-packages/ansible
  ansible collection location = /home/ubuntu/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.1.4
  libyaml = True
```

### Community.general Version

```console
ansible-galaxy collection list community.general

# /usr/local/lib/python3.10/dist-packages/ansible_collections
Collection        Version
----------------- -------
community.general 9.4.0 
```

### OS / Environment

Ubuntu 22.04

### Steps to Reproduce

```yaml
---
- name: Test global inject
  hosts: localhost
  gather_facts: false
  become: true
  tasks:
  - name: ensure application pylint is uninstalled
    community.general.pipx:
      name: pylint
      state: absent
      global: true

  - name: install application pylint
    community.general.pipx:
      name: pylint
      global: true
    register: install_pylint

  - name: inject packages
    community.general.pipx:
      state: inject
      name: pylint
      global: true
      inject_packages:
        - licenses
```

### Expected Results

Expected licenses to be installed in the global venv of pylint.

### Actual Results

```console
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [Test global inject] *********************************************************************************************************************************************************

TASK [ensure application pylint is uninstalled] ***********************************************************************************************************************************
ok: [localhost]

TASK [install application pylint] *************************************************************************************************************************************************
changed: [localhost]

TASK [inject packages] ************************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible_collections.community.general.plugins.module_utils.mh.exceptions.ModuleHelperException
fatal: [localhost]: FAILED! => {"application": {}, "changed": false, "force": false, "index_url": null, "msg": "Trying to inject packages into a non-existent application: pylint", "name": "pylint", "output": {"application": {}, "force": false, "index_url": null, "name": "pylint", "source": null}, "source": null, "vars": {"application": {}, "force": false, "index_url": null, "name": "pylint", "source": null}}

PLAY RECAP ************************************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

ubuntu@ubuntu:~/Desktop/git/pipxtest$ pipx list --global
venvs are in /opt/pipx/venvs
apps are exposed on your $PATH at /usr/local/bin
manual pages are exposed at /usr/local/share/man
   package pylint 3.3.1, installed using Python 3.10.12
    - pylint (symlink missing or pointing to unexpected location)
    - pylint-config (symlink missing or pointing to unexpected location)
    - pyreverse (symlink missing or pointing to unexpected location)
    - symilar (symlink missing or pointing to unexpected location)
ubuntu@ubuntu:~/Desktop/git/pipxtest$ pipx --global list
nothing has been installed with pipx 😴
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibullbot** [comment]
Files identified in the description:
* [`plugins/modules/pipx.py`](https://github.com/ansible-collections/community.general/blob/main/plugins/modules/pipx.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by ansibullbot** [comment]
cc @russoz
[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by russoz** [comment]
Hi @gaby thanks for reporting. I will work on it in the next few days, time allowing. Though by the looks of it, it might be quite simple.

**Comment by gaby** [comment]
@russoz Thank you! I havent tested the other `states`, but anything putting global before the command is probably also failing. The `--global` always has to go after the command `pipx <cmd> --globa`.

**Comment by gaby** [comment]
Probably worth adding an integration test that uses `state: latest` to make sure it properly installs/upgrades using `--global`.

**Comment by gaby** [comment]
I can confirm that "uninstall" also is not working with `global: true`. Probably for the same reason

**Comment by russoz** [comment]
Ok, I got it. The problem was here:
https://github.com/ansible-collections/community.general/blob/main/plugins/modules/pipx.py#L267

I was not passing `global` in the pipx list call. 

However, I was puzzled by this, because I had written tests for install using `global=true`, so I ended noticing that my test was messed up (the pipx 1.7.0 version)  and it was effectively NOT running. Long story short, the PR is going to be somewhat bigger than first anticipated, but I think we'll be in a much better state.
